packages/integration/src/gitlab/SingleInstanceGitlabCredentialsProvider.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GitLabIntegrationConfig } from './config';
+import { GitlabCredentials, GitlabCredentialsProvider } from './types';
+
+export class SingleInstanceGitlabCredentialsProvider
+  implements GitlabCredentialsProvider
+{
+  static create: (
+    config: GitLabIntegrationConfig,
+  ) => GitlabCredentialsProvider = config => {
+    return new SingleInstanceGitlabCredentialsProvider(config.token);
+  };
+
+  constructor(private readonly token?: string) {}
+
+  async getCredentials(): Promise<GitlabCredentials> {
+    return {
+      headers: this.token
+        ? { Authorization: `Bearer ${this.token}` }
+        : undefined,
+      token: this.token,
```

Comment:
Might be worth it to be thorough here, and just do `if (!this.token) { return {}; }` at the top so we don't get keys with undefined in them unnecessarily. Doesn't make any practical difference in al honesty, just a bit of extra ✨